### PR TITLE
👔 refactoring store

### LINKED ISSUE #1308: Refactoring vuex store
- [x] change Vuex design pattern
- [ ] use constants for [mutation](https://vuex.vuejs.org/guide/mutations.html#using-constants-for-mutation-types)
- [x] refactoring to multiple [modules](https://vuex.vuejs.org/guide/modules.html#modules) file (Lang, Fiat, Preferences, Indexer)
- [ ] prepare futur [TS](https://next.vuex.vuejs.org/guide/typescript-support.html) support?
- [x] move fetchIndexer to vuex

### PR DESCRIPTION
**Pull request by roiLeo**
Yeet!

### PR type
- [x] Refactoring Vuex store
- [x] sync collection

### TODO
- [ ] sync GalleryItem v2
- [ ] remove [unused](https://vuex.vuejs.org/guide/plugins.html) vuex plugins
- [ ] remove stuff from layout

### Before submitting Pull Request, please make sure:
- [ ] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried respect high code quality standards
- [ ] I've didn't break any original functionality

### What's new?
- [x] #1308 related 
- [x] #1356 related

### CODE REVIEW

File: components/accounts/utils.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export const toDistribute = (length: number, distribution: number): number => {
 }
 
 export const parseBatchAddresses = (batchAddresses: string): string[] => {
-  const ss58Format: any = store.getters.getChainProperties58Format
+  const ss58Format: any = store.getters['chain/getChainProperties58Format']
```
Comment (vikiival) [inline components/accounts/utils.ts:21]: is this like a good pattern for that ?

File: components/shared/format/Money.vue [hunk 2]
```diff
@@ -41,26 +41,27 @@ export default class Money extends Vue {
   }
 
   private async getFiatValue() {
```
Comment (vikiival) [inline components/shared/format/Money.vue:43]: remove if unused

File: layouts/default.vue [hunk 3]
```diff
@@ -60,83 +58,11 @@ export default class Dashboard extends Vue {
     await cryptoWaitReady()
```
Comment (vikiival) [inline layouts/default.vue:58]: BTW this could be moved into the nuxt/plugins

File: store/chain.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { GetterTree, ActionTree, MutationTree, Commit } from 'vuex'
+
+export const state = () => ({
+  chainProperties: {},
```
Comment (vikiival) [inline store/chain.ts:4]: may add type here ? 
there is just `ss58Format, tokenDecimals, unit`

File: store/indexer.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+          <p class="title is-3">Indexer Error</p>
+          <p class="subtitle">Indexer is not working properly
+          <a target="_blank" rel="noopener noreferrer"  href="https://explorer.subquery.network/subquery/vikiival/magick">and you can check the status here</a></p>
+          <p class="subtitle">If you think this should't happen, report us by
+          <a target="_blank" rel="noopener noreferrer"  href="https://github.com/kodadot/nft-gallery/issues/new?assignees=&labels=bug&template=bug_report.md&title=">creating bug issue with steps to reproduce and screenshot.</a></p>
+        `,
+        type: 'is-danger',
+        hasIcon: true,
+      }
+      this.app.$buefy.snackbar.open(type as any)
```
Comment (vikiival) [inline store/indexer.ts:57]: u sure?

**Review by vikiival — CHANGES_REQUESTED** [review]
I fairly love it.

just small things and i am not sure about this syntax

```ts
store.getters['chain/getChainProperties58Format']
```

can you provide some article?

File: components/accounts/utils.ts (hunk 1 above)
Comment (roiLeo) [reply components/accounts/utils.ts:21]: Oups! I'm used to this [store](https://nuxtjs.org/docs/directory-structure/store/) pattern
```js
  const ss58Format: any = store.getters.chain.getChainProperties58Format
```
is working too, should I change it?

File: store/indexer.ts (hunk 5 above)
Comment (roiLeo) [reply store/indexer.ts]: lol I've tried something here

File: layouts/default.vue (hunk 3 above)
Comment (vikiival) [reply layouts/default.vue:58]: Lets keep it for #1376

File: store/chain.ts (hunk 4 above)
Comment (roiLeo) [reply store/chain.ts]: unit?

File: components/shared/format/Money.vue (hunk 2 above)
Comment (roiLeo) [reply components/shared/format/Money.vue:43]: This could be a nice addition to 'Money' component

File: plugins/pwa.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { NotificationProgrammatic as Notification } from 'buefy'
+
+export default async () => {
+  //@ts-ignore
```
Comment (roiLeo) [inline plugins/pwa.ts:4]: why are you running?

Comment (roiLeo) [reply plugins/pwa.ts:4]: > 

you will need to import some type here

File: store/chain.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    ss58Format: string
+    tokenDecimals: string
+    genesisHash: string
+    tokenSymbol: string
+  }
+}
+
+const defaultState: ChainProperties = {
+  chainProperties: {
+    ss58Format: '',
```
Comment (vikiival) [inline store/chain.ts:14]:
```suggestion
    ss58Format: 42,
```

File: store/chain.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    tokenDecimals: '',
```
Comment (vikiival) [inline store/chain.ts:15]:
```suggestion
    tokenDecimals: 12,
```

File: store/chain.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    genesisHash: '',
+    tokenSymbol: '',
```
Comment (vikiival) [inline store/chain.ts:17]:
```suggestion
    tokenSymbol: 'Unit',
```

File: layouts/centered-half-layout.vue [hunk 10]
```diff
@@ -61,51 +60,11 @@ export default class Dashboard extends Vue {
     await cryptoWaitReady()
     console.log('wasmCrypto loaded')
     this.loadKeyring()
-    this.$store.commit('keyringLoaded')
+    this.$store.commit('keyringLoaded') // TODO: refact to dispatch and move outside
```
Comment (vikiival) [inline layouts/centered-half-layout.vue:63]:
```suggestion
    this.$store.commit('keyringLoaded') // TODO: See #1376
```

**Review by vikiival — COMMENTED** [review]
Really nice cleanup! 
🧹 

this slowpoke moves

**Review by vikiival — APPROVED** [review]

**Comment by yangwao** [comment]
But will it blend 

image

**Pull request merged**